### PR TITLE
Trace more metrics related to the chain tip

### PR DESCRIPTION
**Pull request by intricate**
Related to #801.

Trace metrics for:

- Hash of the tip block.
- Hash of the parent block of the tip block.
- Hash of the tip block issuer's verification key.

### CODE REVIEW

File: cardano-node/src/Cardano/Tracing/Tracers.hs [hunk 1]
```diff
@@ -378,33 +381,65 @@ teeTraceChainTipElide
...
+        traceI :: Integral a => Text -> a -> IO ()
+        traceI msg i = traceNamedObject tr' (meta, LogValue msg (PureI (fromIntegral i)))
+
+    traceD "density"     (fromRational density)
+    traceI "slotNum"     slots
+    traceI "blockNum"    blocks
+    traceI "slotInEpoch" slotInEpoch
+    traceI "epoch"       (unEpochNo epoch)
+    traceNamedObject
+      (appendName "tipBlockHash" tr')
```
Comment (intricate) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:404]: `tipBlockHash` or `lastBlockHash`?

Comment (erikd) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:404]: Whatever fits in best with the rest of the naming.

File: cardano-node/src/Cardano/Tracing/Tracers.hs [hunk 2]
```diff
@@ -378,33 +381,65 @@ teeTraceChainTipElide
...
-  traceI "blockNum"    blocks
-  traceI "slotInEpoch" slotInEpoch
-  traceI "epoch"       (unEpochNo epoch)
+traceChainInformation
+  :: forall blk. ConvertRawHash blk
+  => Trace IO Text
+  -> ChainInformation blk
+  -> IO ()
+traceChainInformation tr chainInfo = do
+    -- TODO this is executed each time the chain changes. How cheap is it?
```
Comment (intricate) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:390]: This is probably more correct:

```suggestion
    -- TODO this is executed each time the tip changes. How cheap is it?
```

File: cardano-node/src/Cardano/Tracing/Tracers.hs (hunk 1 above)
Comment (intricate) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:404]: In that case, probably just `blockHash` and `parentBlockHash` then.

File: cardano-node/src/Cardano/Tracing/Tracers.hs (hunk 2 above)
Comment (erikd) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:390]: Or "chain tip" for extra redundancy :trollface: .

File: cardano-node/src/Cardano/Tracing/HasIssuer.hs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      BlockIssuerVerificationKeyHash
+        . serialiseToRawBytes
+        . verificationKeyHash
+        . StakePoolVerificationKey
+        . toStakePoolKey
+        $ Shelley.bheaderVk bhBody
+    where
+      -- We don't support a "block issuer" key role in @cardano-api@, so we'll
+      -- just convert it to a stake pool key.
+      toStakePoolKey
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Tracing/HasIssuer.hs:66]: Would a `CastVerificationKeyRole` instance make sense here?

**Comment by newhoggy** [comment]
Looks like you need to fix `hlint` errors.

File: cardano-node/src/Cardano/Tracing/HasIssuer.hs (hunk 3 above)
Comment (intricate) [reply cardano-node/src/Cardano/Tracing/HasIssuer.hs:66]: Well we don't have anything like a `BlockIssuerKey` in the API. Maybe we could add one later, if the need arises.

Comment (Jimbo4350) [reply cardano-node/src/Cardano/Tracing/HasIssuer.hs:66]: It would make sense to add it as the need is arising here.

**Comment by newhoggy** [comment]
You have this compiler error:

```
src/Cardano/Node/Protocol/Cardano.hs:77:5: error:
    • No instance for (Cardano.Tracing.HasIssuer.HasIssuer
                         (Ouroboros.Consensus.Shelley.Ledger.Block.ShelleyBlock
                            (Cardano.Ledger.Allegra.AllegraEra StandardCrypto)))
        arising from a use of ‘SomeConsensusProtocol’
    • In the first argument of ‘(<$>)’, namely ‘SomeConsensusProtocol’
      In the expression:
        SomeConsensusProtocol
          <$> mkConsensusProtocolCardano ncb ncs nch files
      In an equation for ‘mkSomeConsensusProtocolCardano’:
          mkSomeConsensusProtocolCardano ncb ncs nch files
            = SomeConsensusProtocol
                <$> mkConsensusProtocolCardano ncb ncs nch files
   |
77 |     SomeConsensusProtocol
   |     ^^^^^^^^^^^^^^^^^^^^^
```

**Comment by intricate** [comment]
@newhoggy: Thanks. Must have come up after a `rebase` on `master`. I'll fix it.

**Review by CodiePP — COMMENTED** [review]
this might still be information that someone finds useful to see in the logs.
is @intricate still on it?

File: cardano-node/src/Cardano/Tracing/Tracers.hs [hunk 4]
```diff
@@ -868,7 +913,7 @@ teeTraceBlockFetchDecision'
     -> Tracer IO (WithSeverity [TraceLabelPeer peer (FetchDecision [Point (Header blk)])])
 teeTraceBlockFetchDecision' tr =
     Tracer $ \(WithSeverity _ peers) -> do
-      meta <- mkLOMeta Info Confidential
+      meta <- mkLOMeta Critical Confidential
```
Comment (newhoggy) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:916]: Why was this changed to `Critical`?

**Comment by newhoggy** [comment]
Is this PR still wanted?

**Comment by intricate** [comment]
It appears that this PR can be closed now that #4915 has been merged.
